views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,532 @@
+                '-ms-transform-origin': '0 0',
+                'transform-origin': '0 0'
+            });
+        },
+
+        /**
+         * position the preview depending on the height of the toolbar
+         * @private
+         */
+        positionPreview: function positionPreview() {
+            var topBarHeight = testRunner.getAreaBroker().getHeaderArea().outerHeight();
+
+            controls.$scaleWrapper.css('top', topBarHeight+ 'px');
+            controls.$scaleWrapper.css('max-height', 'calc(100vh - ' + topBarHeight + 'px )');
+        },
+
+        /**
+         * sets orientation of selected device globally and does some configuration around
+         * @param {String} newOrientation - Device's orientation  could be: 'landscape' or 'portrait'
+         * @private
+         */
+        setOrientation: function setOrientation(newOrientation) {
+            var re;
+            var previewFrame;
+
+            if (newOrientation === orientation) {
+                return;
+            }
+
+            re = new RegExp(orientation, 'g');
+            previewFrame = controls.$scaleWrapper.find('.preview-outer-frame')[0];
+            previewFrame.className = previewFrame.className.replace(re, newOrientation);
+            orientation = newOrientation;
+        },
+
+        /**
+         * Configures screen size configuration state according to passed value from affected control
+         * and calls this.scaleFrame() after all.
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onDeviceChange: function onDeviceChange(event){
+            var element = event.target;
+            var type = selectedDevice;
+            var val = element.value.split(',');
+            var i = val.length;
+            var container = controls.$scaleWrapper.find('.' + type + '-preview-container');
+
+            while (i--) {
+                val[i] = parseFloat(val[i]);
+            }
+            if (type === 'mobile' && orientation === 'portrait') {
+                sizeSettings = {
+                    width: val[1],
+                    height: val[0]
+                };
+            }
+            else {
+                sizeSettings = {
+                    width: val[0],
+                    height: val[1]
+                };
+            }
+
+            if (sizeSettings.width === container.width() && sizeSettings.height === container.height()) {
+                return false;
+            }
+
+            container.css(sizeSettings);
+            this.scaleFrame();
+        },
+
+        /**
+         * Configures selected device's orientation configuration state according to passed value from affected control
+         * and calls this.setOrientation(newOrientation) and this.scaleFrame() after all.
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onOrientationChange: function onOrientationChange(event) {
```

Comment:
Is the event really needed here? Only having the target element would be enough.